src/main/java/org/broadinstitute/hellbender/tools/walkers/sv/EvaluateConcordanceOnNonGenotypedSVVCFs.java

```diff
@@ -0,0 +1,497 @@
+     * @param eval          variant to be evaluated
+     * @param evalInterval  padded interval constructed from {@code eval}
+     * @return              {@code null} if no concordant truth call exist
+     */
+    private SupportingTruth getConcordantTruthCalls(final VariantContext eval, final SVInterval evalInterval) {
+        SupportingTruth bestSupportingTruth = null;
+
+        final Iterator<SVIntervalTree.Entry<List<TruthVariant>>> overlappingTruths = truthTree.overlappers(evalInterval);
+
+        Float maxConcordanceScore = 0f;
+        while (overlappingTruths.hasNext()) {
+            final List<TruthVariant> truthVariants = overlappingTruths.next().getValue();
+            for (int i = 0; i < truthVariants.size(); ++i ) {
+                final TruthVariant truthVariant = truthVariants.get(i);
+                final VariantContext truthVarCtx = truthVariant.getTruth();
+                float concordanceScore = computeConcordanceScore(eval, truthVarCtx, null);
+                final Boolean concordant = variantConcordanceTester.test(eval, truthVarCtx, null);
+                if (concordant && concordanceScore > maxConcordanceScore) {
+                    SupportingTruth supportingTruth = new SupportingTruth(truthVarCtx, concordanceScore);
+                    final TruthVariant updatedTruthVariant =  eval.isFiltered() ? new FilteredFalseNegative(eval, supportingTruth.truthVar, supportingTruth.maxSupportScore)
+                                                                                : new TruePositive(eval, supportingTruth.truthVar, supportingTruth.maxSupportScore);
+                    truthVariants.set(i, updatedTruthVariant);
+                    bestSupportingTruth = supportingTruth;
+                }
+            }
+        }
+        return bestSupportingTruth;
+    }
+
+    // TODO: 6/3/18 test if heuristic overlap-based concordance score is over a hard threshold, in the long run, haplotype concordance should be evaluated instead
+    /**
+     * Assuming argument VariantContext's overlap.
+     */
+    @Override
+    protected TriPredicate<VariantContext, ReferenceContext> getConcordanceTester() {
+        return (eval, truth, r) -> recipOvpFrac <= computeConcordanceScore(eval, truth, r);
+    }
+
+    /**
+     * >= 0.5 is considered concordant.
+     */
+    @Override
+    protected float computeConcordanceScore(final VariantContext eval, final VariantContext truth, final ReferenceContext context) {
+        final String evalType = eval.getAttributeAsString(GATKSVVCFConstants.SVTYPE, "");
+        final String truthType = truth.getAttributeAsString(GATKSVVCFConstants.SVTYPE, "");
+
+        // TODO: 6/3/18 we only consider insertion length but DO NOT YET MATCH inserted sequence (we should, in the long run, but that involves haplotype reconstruction)
+        if (evalType.equals(SimpleSVType.SupportedType.INS.name()) || evalType.equals(SimpleSVType.SupportedType.DUP.name())) {
```

Comment:
What  about SimpleSVType.SupportedType.valueOf(evalType) == SimpleSVType.SupportedType.INS

with some refactoring you could simplify this expression.

Reply:
done with the valueOf. 
Not sure about what type of refactoring is meant?